src/celeritas/ext/detail/GeantPhysicsList.cc

```diff
@@ -178,122 +198,81 @@ void GeantPhysicsList::add_e_processes()
 
-        // Electron
-        auto coulomb_scat_electron = std::make_unique<G4CoulombScattering>();
-        auto coulomb_model_electron
-            = std::make_unique<G4eCoulombScatteringModel>();
-        coulomb_scat_electron->SetMinKinEnergy(msc_energy_limit);
-        coulomb_model_electron->SetLowEnergyLimit(msc_energy_limit);
-        coulomb_model_electron->SetActivationLowEnergyLimit(msc_energy_limit);
-        coulomb_scat_electron->SetEmModel(coulomb_model_electron.release());
-        physics_list->RegisterProcess(coulomb_scat_electron.release(),
-                                      electron);
-
-        // Positron
-        auto coulomb_scat_positron = std::make_unique<G4CoulombScattering>();
-        auto coulomb_model_positron
-            = std::make_unique<G4eCoulombScatteringModel>();
-        coulomb_scat_positron->SetMinKinEnergy(msc_energy_limit);
-        coulomb_model_positron->SetLowEnergyLimit(msc_energy_limit);
-        coulomb_model_positron->SetActivationLowEnergyLimit(msc_energy_limit);
-        coulomb_scat_positron->SetEmModel(coulomb_model_positron.release());
-        physics_list->RegisterProcess(coulomb_scat_positron.release(),
-                                      positron);
+        auto process = std::make_unique<G4CoulombScattering>();
+        auto model   = std::make_unique<G4eCoulombScatteringModel>();
+        process->SetMinKinEnergy(msc_energy_limit);
+        model->SetLowEnergyLimit(msc_energy_limit);
+        model->SetActivationLowEnergyLimit(msc_energy_limit);
+        process->SetEmModel(model.release());
+        physics_list->RegisterProcess(process.release(), p);
 
         CELER_LOG(debug) << "Loaded Coulomb scattering with "
                             "G4eCoulombScatteringModel";
     }
 
-    if (true)
+    if (options_.msc != MscModelSelection::none)
     {
         // Multiple scattering: Urban (low E) and WentzelVI (high E) models
         double msc_energy_limit = G4EmParameters::Instance()->MscEnergyLimit();
-        auto   msc_electron     = std::make_unique<G4eMultipleScattering>();
-        auto   msc_positron     = std::make_unique<G4eMultipleScattering>();
 
-        if (true)
-        {
-            // Urban model
-            // Electron
-            auto urban_msc_electron = std::make_unique<G4UrbanMscModel>();
-            urban_msc_electron->SetHighEnergyLimit(msc_energy_limit);
-            msc_electron->SetEmModel(urban_msc_electron.release());
+        auto process = std::make_unique<G4eMultipleScattering>();
 
-            // Positron
-            auto urban_msc_positron = std::make_unique<G4UrbanMscModel>();
-            urban_msc_positron->SetHighEnergyLimit(msc_energy_limit);
-            msc_positron->SetEmModel(urban_msc_positron.release());
+        if (options_.msc == MscModelSelection::urban)
+        {
+            auto model = std::make_unique<G4UrbanMscModel>();
+            model->SetLowEnergyLimit(msc_energy_limit);
```

Comment:
I think you mistakenly changed this bit with an automatic find/replace. Urban has a high limit threshold, whereas wentzelvi has a low limit one.

Reply:
Ah, that was my error: I didn't notice when updating them in the first place that the two set points were different function calls.

Reply:
Wait, so `msc energy limit` is the boundary *between* two separate MSC models?? What if you only have one or the other enabled?

Reply:
Yes. That value is the boundary and by default is 100 MeV.

> What if you only have one or the other enabled?

Good point. I guess we could call `model->SetHighEnergyLimit(100 * TeV)` or similar when enabling only urban, specially since wentzelvi is more important for precise muon/hadron msc at low angles (and single scattering is used for large angles, as far as I remember). 

@whokion any thoughts on this? Do we have a hardcoded limit on your msc implementation in Celeritas?

Reply:
I discussed this point with @whokion offline. He did not explicitly applied a 100 MeV threshold on the Celeritas side, but the model should effectively work only up to this limit and we should add it in the near future (or build urban msc_params below this limit). 

TLDR: his conclusion is that keeping `model->SetHighEnergyLimit(msc_energy_limit)` is better, as we won't have urban msc enabled above the default 100 MeV.

Reply:
> Use new parameter MscEnergyLimit() (default is 100 MeV) in EM physics
    constructors, allowing to customise transition from low-energy multiple
    scattering to high energy combination of multiple and single scattering.
 
 Ughhh why would you name a *boundary* between two MSC models as an MSC *limit*...

Reply:
Well... a boundary is technically a limit... :grimacing:
But yeah, I totally agree that it's a poorly chosen variable/function naming, and I also got confused the first time I looked into it.

Reply:
it's not a limit for MSC 🙄 if you called it LowEnergyMscUpperLimit or HighEnergyMscUpperLimit you could argue it... ;)